packages/context-drawer/src/ContextDrawer/ContextDrawer.styles.ts

```diff
@@ -0,0 +1,112 @@
+import { css, cx } from '@leafygreen-ui/emotion';
+import { Theme } from '@leafygreen-ui/lib';
+import {
+  addOverflowShadow,
+  borderRadius,
+  color,
+  InteractionState,
+  Side,
+  spacing,
+  Variant,
+} from '@leafygreen-ui/tokens';
+
+import { TRANSITION_DURATION } from '../constants';
+
+const baseOuterContainerStyles = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+export const getOuterContainerStyles = ({
+  className,
+}: {
+  className?: string;
+}) => cx(baseOuterContainerStyles, className);
+
+const baseInnerContainerStyles = css`
+  display: flex;
+  flex-direction: column;
+  border-radius: ${borderRadius[400]}px;
+  border: 1px solid transparent;
+  border-bottom-width: 0;
+  transition: border ${TRANSITION_DURATION}ms ease-in-out;
+`;
+
+const getOpenContainerStyles = ({ theme }: { theme: Theme }) => css`
+  border: 1px solid
+    ${color[theme].border[Variant.Secondary][InteractionState.Default]};
+`;
+
+export const getInnerContainerStyles = ({
+  showBorder,
+  theme,
+}: {
+  showBorder: boolean;
+  theme: Theme;
+}) =>
+  cx(baseInnerContainerStyles, {
+    [getOpenContainerStyles({ theme })]: showBorder,
+  });
+
+const contentWrapperStyles = css`
+  position: relative;
+  overflow: hidden;
+  transition-property: height, opacity;
+  transition-duration: ${TRANSITION_DURATION}ms;
+  transition-timing-function: ease-in-out;
+  height: 0;
+  opacity: 0;
+  visibility: hidden;
+
+  :focus-visible {
+    outline: none;
+  }
+`;
+
+const getExpandedContentStyles = (height: number | string) => {
+  const expandedHeight = typeof height === 'number' ? `${height}px` : height;
```

Comment:
on one hand I like this api, on the other it'll break silently if someone passes `"100"` as `expandedHeight`. Not sure how to remedy that, and probably a fine caveat. Just wanted to call it out in case you had any ideas for a fix